components/calendar/src/hebrew.rs

```diff
@@ -111,19 +112,78 @@ impl Calendar for Hebrew {
 
     fn date_from_codes(
         &self,
-        _era: types::Era,
-        _year: i32,
-        _month_code: types::MonthCode,
-        _day: u8,
+        era: types::Era,
+        year: i32,
+        month_code: types::MonthCode,
+        day: u8,
     ) -> Result<Self::DateInner, CalendarError> {
-        // let year = if era.0 == tinystr!(16, "hebrew") {
-        //     year
-        // } else {
-        //     return Err(CalendarError::UnknownEra(era.0, self.debug_name()));
-        // };
-
-        // ArithmeticDate::new_from_codes(self, year, month_code, day).map(HebrewDateInner)
-        todo!("#3789")
+        let is_leap_year = Self::is_leap_year(year);
+
+        let year = if era.0 == tinystr!(16, "hebrew") {
+            year
+        } else {
+            return Err(CalendarError::UnknownEra(era.0, self.debug_name()));
+        };
+
+        let month_ordinal = match month_code.0.to_string().as_str() {
```

Comment:
Issue: don't call `to_string`.

Suggestion: For readability, I think I'd prefer two match statements instead of one with these different arms. So, more like,

```rust
let month_code_str = month_code.0.as_str();
let month_ordinal = if is_leap_year {
    match month_code_str {
        "M01" => 1,
        "M02" => 2,
        // ...
        "M05" => 5,
        "M05L" => 6,
        "M06" => 7,
        // ...
    }
} else {
    match month_code_str {
        "M01" => 1,
        "M02" => 2,
        // ...
        "M05" => 5,
        "M06" => 6,
        // ...
    }
};
```

Reply:
done